commonmark-ext-heading-anchor/src/main/java/org/commonmark/ext/heading/anchor/HeadingAnchorExtension.java

```diff
@@ -29,20 +29,78 @@
  */
 public class HeadingAnchorExtension implements HtmlRenderer.HtmlRendererExtension {
 
-    private HeadingAnchorExtension() {
+    private final String defaultId;
+    private final String idPrefix;
+    private final String idSuffix;
+
+    private HeadingAnchorExtension(String defaultId, String idPrefix, String idSuffix) {
+        this.defaultId = defaultId;
+        this.idPrefix = idPrefix;
+        this.idSuffix = idSuffix;
     }
 
     public static Extension create() {
-        return new HeadingAnchorExtension();
+        return create(builder());
+    }
+
+    public static Extension create(Builder builder) {
```

Comment:
This does not need to be public. Instead, just pass the builder to the private constructor. See what we do in `HtmlRenderer`.